lib/retrieval-stats.js

```diff
@@ -103,15 +90,18 @@ export const buildRetrievalStats = (measurements, telemetryPoint) => {
  *
  * @param {import('./typings').Point} point
  * @param {string} fieldNamePrefix
- * @param {hdr.Histogram} histogram
+ * @param {number[]} values
  */
-const addHistogramToPoint = (point, fieldNamePrefix, histogram) => {
-  if (histogram.totalCount < 1) return
-  point.intField(`${fieldNamePrefix}_min`, histogram.minNonZeroValue)
-  point.intField(`${fieldNamePrefix}_mean`, histogram.mean)
-  point.intField(`${fieldNamePrefix}_max`, histogram.maxValue)
+const addHistogramToPoint = (point, fieldNamePrefix, values) => {
+  const count = values.length
+  if (count < 1) return
+  values.sort()
+  point.intField(`${fieldNamePrefix}_min`, values[0])
+  point.intField(`${fieldNamePrefix}_mean`, values.reduce((sum, v) => sum + v, 0) / count)
+  point.intField(`${fieldNamePrefix}_max`, values[count - 1])
   for (const p of [10, 50, 90, 95]) {
-    point.intField(`${fieldNamePrefix}_p${p}`, histogram.getValueAtPercentile(p))
+    const ix = Math.floor(p * count / 100)
+    point.intField(`${fieldNamePrefix}_p${p}`, values[ix])
```

Comment:
is this really a percentile? Eg with values `[0, 50, 100]` the 90th percentile should be `90` but with this logic would be `100`. See https://www.calculatorsoup.com/calculators/statistics/percentile-calculator.php

Reply:
Nice! Thank you for sharing a link to the precise calculation of indexes from percentiles.

We currently have more than 700k measurements; I think the rounding error in percentile calculation is negligible.

Let's land my PR as-is and then improve the `percentile to index` calculation in a follow-up pull request.

IMO, the memory savings are so huge that it's okay to have less precise stats for a while.

Reply:
I have already opened a PR, see here: https://github.com/filecoin-station/spark-evaluate/pull/89